src/IdentityServer4.WsFederation/WsFederationMessageParser.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Net;
+using Microsoft.IdentityModel.Protocols.WsFederation;
+
+namespace IdentityServer4.WsFederation
+{
+    public static class WsFederationMessageParser
+    {
+        public static WsFederationMessage GetSignInRequestMessage(string encodedUrl)
+        {
+            var decodedUrl = WebUtility.UrlDecode(encodedUrl);
```

Comment:
this is not necessary WsFederationMessage.FromUri() takes care of things that need to be decoded.